firmware_revamp/can_api/yaml_to_dbc.py

```diff
@@ -0,0 +1,112 @@
+import yaml
+
+# DEFAULT VALUES FOR DBC MESSAGES AND SIGNALS
+ENDIAN = "@0" # big
+SCALE_DEFAULT = 1
+OFFSET_DEFAULT = 0
+UNIT_DEFAULT = "n/a"
+MIN_DEFAULT = 0
+MAX_DEFAULT = 0
+# Acts as a counter that updates by "length" value everytime a signal is parsed
+BIT_START = 0
+NODE_NAME = "Vector__XXX"
+
+def loadYaml(filename):
+    with open(filename) as stream:
+        file_dict = yaml.safe_load(stream)
+    return file_dict
+
+def dbcMessagesTXGenerator(messagesTX_dict, dbc_file):
+    ''' 
+    Input: 
+        Dictionary with all TX messages
+        DBC file to write to   
+    Output: Writes DBC message and signal to DBC file 
+    '''
+    for message in messagesTX_dict:
+        message_dict = messagesTX_dict[message]
+
+        # parse dictionary and set variables
+        MESSAGE_NAME = message
+        CAN_ID = message_dict["ID"]
+        BYTE_LEN = message_dict["Data Bytes"]
+
+        # Create dict of signals
+        signals_dict = message_dict["Signals"]
+
+        # Generate DBC message
+        DBC_message = "BO_ {} {}: {} {}\n".format(CAN_ID, MESSAGE_NAME, BYTE_LEN, NODE_NAME)
+        print(DBC_message)
+        dbc_file.write(DBC_message)
+
+        # Process signals
+        dbcSignalGenerator(signals_dict, dbc_file)
+
+def dbcSignalGenerator(signals_dict, dbc_file):
+    '''
+    Input: DBC
+    '''
+    for signal in signals_dict:
+        signal_param_dict = signals_dict[signal]
+
+        # Parse signal dict and set variables to be used to generate DBC
+        SIGNAL_NAME = signal
+        global BIT_START

```

Comment:
any particular reason why we need this to be global instead of just initializing it as `BIT_START=0` at the top of function?